themes/_core/js/product.js

```diff
@@ -26,126 +26,278 @@ import $ from 'jquery';
-      });
     });
+}
 
-    $.post(event.reason.productUrl, {ajax: '1', action: 'refresh'}, null, 'json').then(function(resp) {
-      $('.product-prices').replaceWith(resp.product_prices);
-      $('.product-customization').replaceWith(resp.product_customization);
-      $('.product-variants').replaceWith(resp.product_variants);
-      $('.product-discounts').replaceWith(resp.product_discounts);
-      $('.images-container').replaceWith(resp.product_cover_thumbnails);
-      $('.product-additional-info').replaceWith(resp.product_additional_info);
-      $('#product-details').replaceWith(resp.product_details);
-
-      // Replace all "add to cart" sections but the quantity input in order to keep quantity field intact i.e.
-      // Prevent quantity input from blinking with classic theme.
-      let $productAddToCart;
-      $(resp.product_add_to_cart).each(function(index, value) {
-          if ($(value).hasClass('product-add-to-cart')) {
-            $productAddToCart = $(value);
-          }
-      });
-      replaceAddToCartSections($productAddToCart);
+/**
+ * Find DOM elements and replace their content
+ * 
+ * @param {object} Replacement Data to be replaced on the current page
+ */
+function replaceAddToCartSection(replacement) {
+    const destinationObject = $(replacement.$targetParent.find(replacement.targetSelector));
+    if (destinationObject.length <= 0) {
+        return;
+    }
+    const replace = replacement.$addToCartSnippet.find(replacement.targetSelector);
 
-      const minimalProductQuantity = parseInt(resp.product_minimal_quantity, 10);
-      const quantityInputSelector = '#quantity_wanted';
-      let quantityInput = $(quantityInputSelector);
-      const quantity_wanted = quantityInput.val();
+    if (replace.length > 0) {
+        destinationObject.replaceWith(replace[0].outerHTML);
+    } else {
+        destinationObject.html('');
+    }
+}
 
-      if (!isNaN(minimalProductQuantity) && quantity_wanted < minimalProductQuantity && eventType !== 'updatedProductQuantity') {
-        quantityInput.attr('min', minimalProductQuantity);
-        quantityInput.val(minimalProductQuantity);
-      }
+/**
+ * @param {object} $container
+ * @param {string} textError
+ */
+function showError($container, textError) {
+    const $error = $(`<div class="alert alert-danger ajax-error" role="alert">${textError}</div>`);
+    $container.replaceWith($error);
+}
 
-      if (event.refreshUrl) {
-        window.history.pushState({id_product_attribute: resp.id_product_attribute}, undefined, resp.product_url);
-      }
+$(document).ready(function () {
+    // Listen on all form elements + those who have a data-product-attribute
+    $('body').on(
+        'change touchspin.on.startspin',
+        '.product-variants *[name]',
+        function(e) {
+            prestashop.emit('updateProduct', {
+                eventType: 'updatedProductCombination',
+                event: e
+            });
```

Comment:
The data sent to this object has changed. Themes relying on it will likely fail if they can't find the expected properties. Even if it works, we can't accept it because of the compatibility issues.

```js
          reason: {
            productUrl: resp.productUrl
          },
          refreshUrl: $productRefresh.data('url-update'),
          eventType: eventType,
          resp: resp
```

@eternoendless, @mickaelandrieu?